### PR TITLE
Add SAI_INGRESS_PRIORITY_GROUP_STAT_DROPPED_PACKETS counter, create new FlexCounter group

### PR DESCRIPTION
**Pull request by ayurkiv-nvda**
…

Signed-off-by: Andriy Yurkiv <ayurkiv@nvidia.com>

**What I did**
I added new FlexCounter group for SAI_INGRESS_PRIORITY_GROUP_STAT_DROPPED_PACKETS counter

**Why I did it**
Added appropriate logic to sonic-swss

**How I verified it**
check Flex counter group via redis-cli :

```
admin@arc-switch1041:~$ redis-cli -n 5
127.0.0.1:6379[5]> KEYS *FLEX_COUNTER_TABLE:PG_DROPPED_PACKETS_STAT_COUNTER*
...............
127.0.0.1:6379[5]> HGETALL FLEX_COUNTER_TABLE:PG_DROPPED_PACKETS_STAT_COUNTER:oid:0x1a0000000000c7
1) "PG_COUNTER_ID_LIST"
2) "SAI_INGRESS_PRIORITY_GROUP_STAT_DROPPED_PACKETS"
```
check counters:
```
admin@arc-switch1041:~$ redis-cli -n 2
127.0.0.1:6379[2]> HGETALL COUNTERS_PG_NAME_MAP
127.0.0.1:6379[2]> HGETALL COUNTERS:oid:0x1a000000000062
1) "SAI_INGRESS_PRIORITY_GROUP_STAT_XOFF_ROOM_WATERMARK_BYTES"
2) "0"
3) "SAI_INGRESS_PRIORITY_GROUP_STAT_SHARED_WATERMARK_BYTES"
4) "0"
5) "SAI_INGRESS_PRIORITY_GROUP_STAT_DROPPED_PACKETS"
6) "0"
```
**Details if related**

### CODE REVIEW

**Comment by volodymyrsamotiy** [comment]
Please check if we can add VS test.
Maybe something similar to what we have for "watermark" counters?
https://github.com/Azure/sonic-swss/blob/master/tests/test_watermark.py

File: orchagent/portsorch.cpp [hunk 1]
```diff
@@ -4159,7 +4174,19 @@ void PortsOrch::generatePriorityGroupMapPerPort(const Port& port)
...
+        m_flexCounterTable->set(key, fieldValues);
 
+        delimiter = "";
+        std::ostringstream ingress_pg_drop_packets_counters_stream;
+        key = getPriorityGroupDropPacketsFlexCounterTableKey(id);
+        /* Add dropped packets counters to flex_counter */
+        for (const auto& it: ingressPriorityGroupDroppedPktsStatIds)
+        {
+            ingress_pg_drop_packets_counters_stream << delimiter << sai_serialize_ingress_priority_group_stat(it);
+            delimiter = comma;
```
Comment (nazariig) [inline orchagent/portsorch.cpp]: @ayurkiv-nvda this is ugly way, even though it's commonly used in SONiC.
What about this one:
```
for (const auto& it: ingressPriorityGroupDroppedPktsStatIds)
{
    ingress_pg_drop_packets_counters_stream << delimiter << sai_serialize_ingress_priority_group_stat(it);
    if (delimiter.empty())
    {
        delimiter = comma;
    }
}
```

File: orchagent/portsorch.cpp [hunk 2]
```diff
@@ -56,7 +56,7 @@ extern Directory<Orch*> gDirectory;
 #define QUEUE_WATERMARK_FLEX_STAT_COUNTER_POLL_MSECS "10000"
 #define PG_WATERMARK_FLEX_STAT_COUNTER_POLL_MSECS    "10000"
 #define PORT_RATE_FLEX_COUNTER_POLLING_INTERVAL_MS   "1000"
-
+#define INGRESS_PG_STAT_DROPPED_PACKETS_POLLING_INTERVAL_MS "10000"
```
Comment (nazariig) [inline orchagent/portsorch.cpp]: @ayurkiv-nvda how about this (group them together):
```
#define PG_WATERMARK_FLEX_STAT_COUNTER_POLL_MSECS    "10000"
#define PG_DROPPED_PACKETS_FLEX_STAT_COUNTER_POLL_MSECS "10000"
```

File: orchagent/flexcounterorch.cpp [hunk 3]
```diff
@@ -30,6 +30,7 @@ unordered_map<string, string> flexCounterGroupMap =
     {"RIF", RIF_STAT_COUNTER_FLEX_COUNTER_GROUP},
     {"RIF_RATES", RIF_RATE_COUNTER_FLEX_COUNTER_GROUP},
     {"DEBUG_COUNTER", DEBUG_COUNTER_FLEX_COUNTER_GROUP},
+    {"INGRESS_PG_STAT_DROP", PG_DROPPED_PACKETS_STAT_COUNTER_FLEX_COUNTER_GROUP},
```
Comment (nazariig) [inline orchagent/flexcounterorch.cpp]: @ayurkiv-nvda how about `INGRESS_PG_STAT_DROP` -> `PG_STAT_DROP/PG_DROP`

File: orchagent/portsorch.cpp (hunk 1 above)
Comment (ayurkiv-nvda) [reply orchagent/portsorch.cpp]: Done

File: orchagent/flexcounterorch.cpp (hunk 3 above)
Comment (ayurkiv-nvda) [reply orchagent/flexcounterorch.cpp]: Done

File: orchagent/portsorch.cpp (hunk 2 above)
Comment (ayurkiv-nvda) [reply orchagent/portsorch.cpp]: Done

File: orchagent/flexcounterorch.cpp [hunk 4]
```diff
@@ -26,6 +26,7 @@ unordered_map<string, string> flexCounterGroupMap =
     {"PFCWD", PFC_WD_FLEX_COUNTER_GROUP},
     {"QUEUE_WATERMARK", QUEUE_WATERMARK_STAT_COUNTER_FLEX_COUNTER_GROUP},
     {"PG_WATERMARK", PG_WATERMARK_STAT_COUNTER_FLEX_COUNTER_GROUP},
+    {"PG_STAT_DROP", PG_DROPPED_PACKETS_STAT_COUNTER_FLEX_COUNTER_GROUP},
```
Comment (nazariig) [inline orchagent/flexcounterorch.cpp]: @ayurkiv-nvda PG_DROP

Comment (ayurkiv-nvda) [reply orchagent/flexcounterorch.cpp]: Done

**Review by nazariig — APPROVED** [review]

**Pull request closed without merging**